src/core/opamParallel.ml

```diff
@@ -383,8 +383,12 @@ module MakeGraph (X: VERTEX) = struct
       let vertex_to_json (i, v) = (string_of_int i, X.to_json v) in
       `O (List.map vertex_to_json vertex_map) in
     let edges =
-      let vertex_inv_map = List.map (fun (i, v) -> (v, i)) vertex_map in
-      let index v = List.assoc v vertex_inv_map in
+      let module VertexMap = Map.Make(Vertex) in
+      let vertex_inv_map =
+        List.fold_left (fun m (i, v) -> VertexMap.add v i m)
```

Comment:
@AltGr can `vertex_map` contain duplicated `v` ? If so, I believe the correct fix is to use `List.fold_right` to get the correct map out of it.

Reply:
I don't think that's possible: `vertex_map` is defined above as the contents of a set of vertices; this guarantees that there are no duplicates.